cli/command/utils.go

```diff
@@ -72,12 +75,21 @@ func PrettyPrint(i any) string {
 	}
 }
 
-// PromptForConfirmation requests and checks confirmation from user.
+type PromptError error
+
+var ErrPromptTerminated = PromptError(errors.New("prompt terminated"))
+
+// PromptForConfirmation requests and checks confirmation from the user.
 // This will display the provided message followed by ' [y/N] '. If
-// the user input 'y' or 'Y' it returns true other false.  If no
+// the user input 'y' or 'Y' it returns true otherwise false.  If no
 // message is provided "Are you sure you want to proceed? [y/N] "
 // will be used instead.
-func PromptForConfirmation(ins io.Reader, outs io.Writer, message string) bool {
+// If the user terminates the CLI with SIGINT or SIGTERM while
+// the prompt is active, the prompt will return false with an ErrPromptTerminated error.
+// When the prompt returns an error, the caller should propagate the error up the stack
+// ultimately closing the io.Reader used for the prompt which will prevent the background
+// goroutine from blocking indefinitely.
```

Comment:
Maybe we can clean up this comment a bit, and also format better with the above? Something like
```go
// PromptForConfirmation requests and checks confirmation from the user.
// This will display the provided message followed by ' [y/N] '. If the user input
// 'y' or 'Y' it returns true otherwise false.  If no message is provided, "Are you
// sure you want to proceed? [y/N] " will be used instead.
//
// If the user terminates the CLI with SIGINT or SIGTERM while the prompt is active,
// the prompt will return false with an ErrPromptTerminated error. When the prompt
// returns an error, the caller should propagate the error up the stack and close the
// io.Reader used for the prompt which will prevent the background goroutine from
// blocking indefinitely.
```